osu.Game/Rulesets/UI/DrawableRuleset.cs

```diff
@@ -194,11 +194,16 @@ private void load(CancellationToken? cancellationToken)
 
             Audio = audioContainer;
 
-            if ((ResumeOverlay = CreateResumeOverlay()) != null)
+            if (UseResumeOverlay)
             {
-                AddInternal(CreateInputManager()
-                    .WithChild(CreatePlayfieldAdjustmentContainer()
-                        .WithChild(ResumeOverlay)));
+                if ((ResumeOverlay = CreateResumeOverlay()) == null)
+                    UseResumeOverlay = false;
```

Comment:
@peppy I'm not getting this. You've mentioned an edge case when the flag is set before BDL, but... why does it matter? What's the fail case?

If the flag is set to false before BDL, then the overlay will be created and added to the hierarchy - but it *should* never be shown, because the `RequestResume()` check tests both the presence of the resume overlay and the value of the flag - the former may be non-null, but the flag is still false?

Is all this just for optimisation? What am I missing here? I wouldn't bother asking but I find it really weird that the drawable rulesest is setting a "config" bool flag on self - and even then, what does that achieve if you could, for instance, set it back to `true` at any point post-load? And if you did, then this implementation is arguably *less* correct, because as the flag is only tested once in BDL, any value changes to it post-BDL effectively don't matter, which make it a property whose behaviour is dependent on timing?

I've also checked out one commit back (https://github.com/ppy/osu/pull/22338/commits/9e04a36d86c9a19e37cf470aa4c4505fa9ad4ef9) and the test still passes, which adds to my confusion.

Reply:
There's no test coverage of this, but the previous logic would set `UseResumeOverlay` to `true` as long as `CreateResumeOverlay` could be created.

My point is that if there's a future *non-mod* usage that sets the property before `LoadComponent`, it would fail and be overwritten by the code in `load()`.

ie.

```csharp
var ruleset = new DrawableOsuRuleset()
{
    UseResumeOverlay = false
};

LoadComponent(ruleset);

Debug.Assert(!ruleset.UseResumeOverlay); // would have failed
```

Reply:
Thanks, I missed that part. For what it's worth I also find the previous variant of that code weird, and don't really get why the creation of the overlay *must* somehow intermingle with the value of the flag in either direction, so I guess I'll need to go stare at it some more and try to find out...

Reply:
I believe that part is just an optimisation, or more a correctness factor (a variable called `UseResumeOverlay` should probably not be `true` if there's no resume overlay). I'm open to removing that cross-talk if you prefer.

Reply:
> I'm open to removing that cross-talk if you prefer.

I think I'd prefer that. I don't find `UseResumeOverlay` having no effect if no overlay was created _that_ weird - definitely not as weird as "spooky action at a distance" timing-dependent flag value changes.

Reply:
I've adjusted, and also removed the shortcutting when `UseResumeOverlay` is false. Let me know if you agree with that.